src/login.js

```diff
@@ -3,43 +3,91 @@ import { Checkbox } from "react-icheck";
 import "./Login.css";
 
 class Login extends  Component{
+  constructor(props){
+    super(props);
+  }
+  
+  renderUserCredentialsEnabled(){
+    if(this.props.userCredentialsEnabled){
+      return (
+        <form action="#" method="post">
+          <div className="form-group has-feedback">
+            <input type="email" className="form-control" placeholder="Email"></input>
+            <span className="glyphicon glyphicon-envelope form-control-feedback"></span>
+          </div>
+          <div className="form-group has-feedback">
+            <input type="password" className="form-control" placeholder="Password"></input>
+            <span className="glyphicon glyphicon-lock form-control-feedback"></span>
+          </div>
+          <div className="row">
+            <div className="col-xs-8">
+              <div className="remember_me">
+                <Checkbox
+                  checkboxClass="icheckbox_square-blue"
+                  increaseArea="50%"
+                  label="Remember Me"
+                />
+              </div>
+            </div>
+            <div className="col-xs-4">
+              <button type="submit" className="btn btn-primary btn-block btn-flat">Sign In</button>
+            </div>
+          </div>
+        </form>
+      );
+    } else {
+      return (<div></div>);
+    }
+  }
+  
+  renderAuthProviders(){
+    let authProviders = [];
+    if(this.props.authProviders.length > 0){
+      authProviders = this.props.authProviders.map(provider => {
+        switch (provider){
+          case "facebook":
+            return (
+              <a href="#" className="btn btn-block btn-social btn-facebook btn-flat">Name
+                <i className="fa fa-facebook"></i> Name
+                Sign in using Facebook
+              </a>
+            );
+          case "google":
+            return (
+              <a href="#" className="btn btn-block btn-social btn-google btn-flat">Name
+                <i className="fa fa-google-plus"></i> Sign in using
+                Google+
+              </a>
+            );
+          default:
```

Comment:
Can we store this information in object elsewhere like:
```
let providers = {
  facebook: {title:'Sign in using Facebook'},
  google: {title:'Sign in using Google+'},
}
```

and iterate this project?

Reply:
I thought that you want to store in an array, but if store it in an object, that also is OK.

Reply:
Object is just to easily reference the value. You specify array of providers too. Let's say..

```
let providers = ['facebook','google']

let providersInfo  = {
  facebook: {title:'Sign in using Facebook'},
  google: {title:'Sign in using Google+'},
}

content = providers.forEach(providerCode => {
  let provider = providersInfo[providerCode]
  return (
    <a href="#" className="btn btn-block btn-social btn-google btn-flat">
       {providersInfo[provider].title}
    </a>
  )
})

```

just example :)